docs/simbad/query_tap.rst

```diff
@@ -0,0 +1,274 @@
+`~astroquery.simbad.SimbadClass.query_tap` (for Table Access Protocol) is the one
+query to rule them all. It allows to interrogate all the information in SIMBAD with the
+Astronomical Data Query Language (ADQL). ADQL is a flavor of the Structured
+Query Language (SQL) adapted to astronomy. To learn more about this language,
+see the `ADQL documentation <https://ivoa.net/documents/ADQL/index.html>`__
+or the `Simbad ADQL cheat sheet <http://simbad.cds.unistra.fr/simbad/tap/help/adqlHelp.html>`__.
+
+Structure of an ADQL query
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The method `~astroquery.simbad.SimbadClass.query_tap` is called with a string containing the
+ADQL query.
+
+.. code-block:: SQL
+
+  /*ADQL queries start with selecting the columns that will be in the output. Usually,
+  the column name is sufficient. If there is a need to lift ambiguity, add the table
+  name first (table_name.column_name). This is also where the number of rows is fixed
+  (here 5).*/
+  SELECT TOP 5 basic.ra, basic.dec, main_id, nbref
+  /*Then comes the declaration of the tables to be included in the query. Here *basic* and
+  *ids*. Their common column is *oid* in *basic* and *oidref* in *ident*.*/
+  FROM basic JOIN ident ON basic.oid = ident.oidref
+  /*The conditions come after. This query filters otypes that are not in any star
+  category, specific redshifts, and the results should have an NGC name in their
+  list of names.*/
+  WHERE (otype != 'star..') AND (rvz_redshift < 1) AND (id LIKE 'NGC%')
+  /*The result is then sorted so that the top 5 selected corresponds to
+  the objects cited by the largest number of papers.*/
+  ORDER BY nbref DESC
+
+This ADQL query can be called with `~astroquery.simbad.SimbadClass.query_tap`: 
+
+.. a bit long due to the ordering
```

Comment:
can we make it faster, e.g. the ordering is not really needed here if it makes it much slower?

Reply:
I could remove it, but I wanted that first query to present every TAP step (select, from, where, order by). The doctest output would also need to be skipped because the output will change very often. Is it a big issue that this one is skipped?

Reply:
no, in this case it's OK to skip it then

Reply:
Actually, timing it with 

```python
import timeit
timeit.timeit("""Simbad.query_tap("SELECT TOP 5 basic.ra, basic.dec, main_id, nbref FROM basic JOIN ident ON basic.oid = ident.oidref WHERE ((otype != 'Cl*..') AND (rvz_redshift < 1) AND (id LIKE 'NGC%')) ORDER BY nbref DESC")""", number=1, setup="from astroquery.simbad import Simbad")
```

Gives 1.04s on my machine. It's slower than the average query (~0.15s) but not dramatic.

Reply:
Oh, that's not slow at all, at least we have much bigger problems to clear out. 

As we don't run the remote test for PRs, I would indeed suggest to change the skip to an ignore output, so we would have a chance of picking up errors if/when it stops working.